controllers/Frontend.php

```diff
@@ -1,61 +1,133 @@
-        require "views/layout.php";
+        $manager_article = new ArticleManager();
+        $posts = $manager_article->get_posts_blog();
+        
+        $manager_user = new UserManager();
+        $admins = $manager_user->get_posts_blog_admin();
+        
+        $myView = new View('home', 'frontend');
+        $myView->render(array('posts' => $posts, 'admins' => $admins));
     }
 
     public function blog()
     {
-        $model_article = new Article();
-        $model_comment = new Comment();
+        $manager_article = new ArticleManager();
+        $posts = $manager_article->get_posts_blog_all();
         
-        $page="frontend/blog";
-        $topbar="frontend/topbar";
-        require "views/layout.php";
+        $manager_user = new UserManager();
+        $admins = $manager_user->get_posts_blog_all_admin();
+        
+        $myView = new View('blog', 'frontend');
+        $myView->render(array('posts' => $posts, 'admins' => $admins));
     }
 
     public function post()
     {
-        $model_article = new Article();
-        $model_comment = new Comment();
-                
-        $page="frontend/post";
-        $topbar="frontend/topbar";
-        require "views/layout.php";
+        if(filter_var(filter_input(INPUT_GET,'id')))
+        {
+            $id = filter_input(INPUT_GET, 'id');
+            $manager = new ArticleManager();
+            $posts = $manager->get_article_blog();
+            
+            $manager_user = new UserManager();
+            $admins = $manager_user->get_article_blog_admin();
+            
+            $manager_comment = new CommentManager();
+            $responses = $manager_comment->get_comments_blog();
+            
+            if(filter_has_var(INPUT_POST, 'submit')){
+            $name = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'name')));
+            $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email')));
+            $comment = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'comment')));
+            $errors = [];
+
+            if(empty($name) || empty($email) || empty($comment)){
+                $errors['empty'] = "Tous les champs n'ont pas été remplis";
+            }else{
+                if(!filter_var($email, FILTER_VALIDATE_EMAIL)){
+                    $errors['email'] = "L'adresse email n'est pas valide";
+                }
+                if(strlen($name) < 5){
+                    $errors['name'] = "Votre nom doit contenir au moins 5 caractères.";
+                }
+                if(strlen($comment) < 5){
+                    $errors['comment'] = "Votre message doit contenir au moins 5 caractères.";
+                }
+            }
+
+            if(empty($errors)){
+                $manager_comment->insert_comment($name,$email,$comment);
+            }
+            }
+            else
+                $errors =[];
+        }
+
+        if($posts == false){
+            header("Location:/page-erreur");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function header() is discouraged](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)